### ISSUE TITLE
forcealtas2 layout does not seem to be working.

### ISSUE THREAD

**Issue by hsparra** [bokeh/datashader#485]
The results of the `forcealtas2_layout` do not match expectations based upon network force directed layout. Also, the `circle_layout` produces a very similar image.

Using PCAP example data
```
df = forceatlas2_layout(nodes_orig, edges_df)
fda2_img = tf.shade(cvs.line(df[1:], 'x', 'y'), cmap=fire)
tf.set_background(fda2_img, color='black’)
```
forceatlas2 results
![datashader_layouts 2017-10-01 17-03-26](https://user-images.githubusercontent.com/7515105/31059669-c2083946-a6cb-11e7-9877-6aae62193ad6.png)

circle_layout results
![datashader_layouts 2017-10-01 17-08-28](https://user-images.githubusercontent.com/7515105/31059665-b2ac762e-a6cb-11e7-97b0-394407e9ebb6.png)

**Comment by hsparra** [comment]
I have it working for circles:

```
positioned_df = circular_layout(nodes_orig, edges_df)
positioned_df = positioned_df.set_index('id')
direct_circle = directly_connect_edges(positioned_df, edges_df)
cir_direct = hv.Curve(direct_circle)
datashade(cir_direct)
```
![datashader_layouts 2017-10-01 20-03-22](https://user-images.githubusercontent.com/7515105/31061160-9bd1606e-a6e3-11e7-9dc9-458d8356b218.png)

**Comment by hsparra** [comment]
ForceAtlas2 still looks off:
```
fa_positioned_df = forceatlas2_layout(nodes_orig, edges_df)
fa_positioned_df = fa_positioned_df.set_index('id')
fa_edges = directly_connect_edges(fa_positioned_df, edges_df)
fa_direct = hv.Curve(fa_edges)
datashade(fa_direct)
```
![datashader_layouts 2017-10-01 20-05-52](https://user-images.githubusercontent.com/7515105/31061200-0b370350-a6e4-11e7-8126-09722d8c42cb.png)

**Comment by hsparra** [comment]
For reference here is the networkX for directed version:
![datashader_layouts 2017-10-01 14-30-01](https://user-images.githubusercontent.com/7515105/31061209-33b3160c-a6e4-11e7-8a4d-185332b4897e.png)

**Comment by philippjfr** [comment]
As far as I can tell the algorithm still doesn't work well. Here's the new result after 10 iterations:

![image](https://user-images.githubusercontent.com/1550771/31172576-03f80386-a8fc-11e7-80b2-14ca45943684.png)

and here's it after 20 iterations:

![image](https://user-images.githubusercontent.com/1550771/31172581-0b42a5f6-a8fc-11e7-98ca-40f0d5fcb022.png)

**Comment by hsparra** [comment]
The 10 iterations is looking promising. Can you run it with 50 or 100 iterations?

**Comment by philippjfr** [comment]
I've already done so, it looks pretty much like the 20 iterations above only worse.

**Comment by hsparra** [comment]
Do they collapse down into a big ball? That is what it looks like is happening with the 20 iterations, but I cannot tell if it is that or some are being thrown farther out.

**Comment by philippjfr** [comment]
That's right, most nodes end up in a ball in the center.

**Comment by hsparra** [comment]
Sounds like the "gravity" parameter is dominating the repulsion parameter.

**Comment by jbednar** [comment]
Presumably this should be addressed by comparing implementations with the same parameter values and the same data?

**Comment by jbcrail** [comment]
The main difference between the implementations is in how they convert their respective input to a sparse matrix. I've narrowed it down to `_convert_graph_to_sparse_matrix`. I'm working with equivalent data at the moment.

**Comment by jbednar** [comment]
I think this issue was fixed by #488 and possibly another later PR.
